saleor/discount/utils.py

```diff
@@ -38,15 +47,17 @@ def add_voucher_usage_by_customer(voucher, customer_email):
     VoucherCustomer.objects.create(voucher=voucher, customer_email=customer_email)
 
 
-def remove_voucher_usage_by_customer(voucher, customer_email):
+def remove_voucher_usage_by_customer(voucher: "Voucher", customer_email: str):
     voucher_customer = VoucherCustomer.objects.filter(
         voucher=voucher, customer_email=customer_email
     )
     if voucher_customer:
         voucher_customer.delete()
 
 
-def get_product_discount_on_sale(product, product_collections, discount: DiscountInfo):
+def get_product_discount_on_sale(
+    product: "Product", product_collections: Set["Collection"], discount: DiscountInfo
+):
```

Comment:
`-> Money` AFAIK

Reply:
Thats why we need typing ;)

 ```
    product_discount = get_product_discount_on_sale(variant.product, set(), discount)
    discounted_price = product_discount(product.price)
```

Reply:
Honestly I dont know whats the type of this right now. Covering this module is out of scope of this PR anyway.